pkg/machinery/client/client.go

```diff
@@ -182,7 +183,18 @@ func (c *Client) GetConn(ctx context.Context, opts ...grpc.DialOption) (*grpc.Cl
 		// Such possibilities include SRV records, multiple IPs from A and/or AAAA
 		// records, and descriptive TXT records which include things like load
 		// balancer specs.
-		target = fmt.Sprintf("dns:///%s:%d", net.FormatAddress(endpoints[0]), constants.ApidPort)
+		endpoint := endpoints[0]
```

Comment:
this is the wrong place actually. 

we need to handle it same way for the case when `len(endpoints) > 1` and `len(endpoints) == 1`